packages/koehnlab/spin_hamiltonians/spin_systems.py

```diff
@@ -0,0 +1,333 @@
+import numpy as np 
+from .phys_const import ge, muNbohr
+from .spin_utils import spinMat
+
+
+numThr = 1e-8
+maxDim = 1000 
+
+class spin:
+
+    def __init__(self,S,type="e",gnuc=None):
```

Comment:
`type` is superfluous. You could simply check if `gnuc` is `None` or not. If you want to keep `type` for possible future extensions, creating an `enum` for the possible values will greatly improve usability of the function (the set of allowed values is immediately clear and typos can be caught by IDEs and not only at runtime)

Reply:
I thought that giving the type is a bit more explicit; actually, I was originally thinking of giving the isotope here to set gnuc automatically. I guess this feature is still in the flow ... I did not look up how to implement an enum yet, can you add this quickly?